src/LogicTasks/Semantics/Decide.hs

```diff
@@ -61,14 +62,15 @@ instance Show Choice where
 genDecideInst :: DecideConfig -> Gen DecideInst
 genDecideInst DecideConfig{..} = do
     let percentTrueEntries' = fromMaybe (0, 100) percentTrueEntries
-
+    -- jscpd:ignore-start
     formula <- case formulaConfig of
       (FormulaArbitrary syntaxTreeConfig) ->
-        InstArbitrary <$> genSynTree syntaxTreeConfig  `suchThat` withRatio percentTrueEntries'
+        InstArbitrary <$> genSynTree syntaxTreeConfig  `suchThat` \t -> withRatio percentTrueEntries' t && synTreeDependsOnAllAtomics t
       (FormulaCnf cnfCfg) ->
-        InstCnf <$> genCnf' cnfCfg `suchThat` withRatio percentTrueEntries'
+        InstCnf <$> genCnf' cnfCfg `suchThat` \cnf -> withRatio percentTrueEntries' cnf && cnfDependsOnAllAtomics cnf
       (FormulaDnf dnfCfg) ->
-        InstDnf <$> genDnf' dnfCfg `suchThat` withRatio percentTrueEntries'
+        InstDnf <$> genDnf' dnfCfg `suchThat` \dnf -> withRatio percentTrueEntries' dnf && dnfDependsOnAllAtomics dnf
+    -- jscpd:ignore-end
```

Comment:
Shouldn't it be possible to pull the three separate "depends on" checks outside the `case`, by using `formulaDependsOnAllAtoms` from `src/Util.hs` defined further below?

Reply:
Was ich konkret meine, ist dass die Änderungen in den drei Branches zurückgenommen werden könnten, und dafür ein `flip suchThat formulaDependsOnAllAtoms $` vor das `case` geschrieben.